Tests/OptimizelyTests-DataModel/AttributeValueTests.swift

```diff
@@ -338,48 +342,150 @@ extension AttributeValueTests {
         XCTAssert(model.stringValue == "UNKNOWN")
     }
     
-    func testIsExactMatchWithInvalid() {
-        let attr = AttributeValue.string("string")
+    func testIsExactMatchWithInvalidValue() {
+        let attr = AttributeValue.others
+        var tmpError: Error?
+        do {
+            _ = try attr.isExactMatch(with: invalidValue, condition: conditionString, name: attributeKey)
+        } catch {
+            tmpError = error
+        }
+        XCTAssertEqual("[Optimizely][Error] " + tmpError!.localizedDescription, OptimizelyError.evaluateAttributeInvalidCondition(conditionString).localizedDescription)
+    }
+    
+    func testIsExactMatchWithInfiniteValue() {
+        let attr = AttributeValue.double(Double.infinity)
         var tmpError: Error?
         do {
-            _ = try attr.isExactMatch(with: ["invalidType"])
+            _ = try attr.isExactMatch(with: invalidValue, condition: conditionString, name: attributeKey)
         } catch {
             tmpError = error
         }
-        XCTAssertTrue(tmpError != nil)
+        XCTAssertEqual("[Optimizely][Error] " + tmpError!.localizedDescription, OptimizelyError.evaluateAttributeInvalidCondition(conditionString).localizedDescription)
     }
     
-    func testIsGreaterWithInvalid() {
+    func testIsExactMatchWithInvalidAttributeValue() {
         let attr = AttributeValue.string("string")
         var tmpError: Error?
         do {
-            _ = try attr.isSubstring(of: ["invalidType"])
+            _ = try attr.isExactMatch(with: invalidValue, condition: conditionString, name: attributeKey)
         } catch {
             tmpError = error
         }
-        XCTAssertTrue(tmpError != nil)
+        XCTAssertEqual("[Optimizely][Error] " + tmpError!.localizedDescription, OptimizelyError.evaluateAttributeInvalidType(conditionString, invalidValue, attributeKey).localizedDescription)
```

Comment:
These tests validate expected error "type"s, but not testing if the log messages are actually printed out.
We throw errors. At some point, SDK is expected to catch it and prints out the log message. 
We need validate that the expected error messages are actually logged.

Reply:
Hi jae, I have added new tests in decision service to verify if thrown errors are logged properly using `MockLogger`. i have also kept the newly added tests for `UserAttribute` and `AttributeValue` to test error handling but have removed assertion of log messages from them.